### ISSUE TITLE
chunks  when training the model

### ISSUE THREAD

**Issue by TheRainInSpain** [shilab/STI#10]
Hi, developers,

    I would like to ask a question that about the training strategy. 
    I have read the code and have a question on it. In file `STICI_V1.1.py`, line 1130-1136 was

```
 for w in range(len(break_points) - 1):
        if chunks_done[w]:
            pprint(f"Skipping chunk {w + 1}/{len(break_points) - 1} due to previous training.")
            continue
        if args.which_chunk != -1 and w + 1 != args.which_chunk:
            pprint(f"Skipping chunk {w + 1}/{len(break_points) - 1} due to your request using --which-chunk.")
            continue
```

and line 1172-1182 was:

```
with strategy.scope():
            model = create_model(model_args)
            history = model.fit(train_dataset, steps_per_epoch=steps_per_epoch,
                                epochs=NUM_EPOCHS,
                                validation_data=valid_dataset,
                                validation_steps=validation_steps,
                                callbacks=callbacks, verbose=args.verbose)
            model.save(f"{args.save_dir}/models/w_{w}.ckpt")
            # tf.saved_model.save(model, f"{args.save_dir}/models/w_{w}.keras")
            chunks_done[w] = True
            save_chunk_status(args.save_dir, chunks_done)
```

It seems that when processing each chunks, it trained a model, and same the model. And then when doing the imputation on test set, it used the corresponding model to impute? And if the sequences are very long and the window size is small, will there be a huge amount of models?

Thanks a lot for your help.

**Comment by ErfanMowlaei** [comment]
@TheRainInSpain That is correct. The code breaks the entire region into chunks (not to confuse with windows inside the model) and trains one model per chunk. These models are used later on for prediction on the same chunk. This strategy is in place for any imputation model we have at hand already (e.g., Minimac, Beagle, etc.,)
